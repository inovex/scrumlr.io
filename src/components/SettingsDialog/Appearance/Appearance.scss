@import "constants/style";
@import "../SettingsDialog";

.appearance-container {
  display: flex;
  flex-direction: column;
  gap: 32px;
  padding-bottom: $settings-dialog-container--bottom;
  overflow-y: auto;
  flex-grow: 1;

  @include scrollbar();
}

.appearance-container > button {
  flex: 0;
}

.appearance-settings_language-dropdown {
  justify-content: space-between;
  padding-right: 24px;
}
