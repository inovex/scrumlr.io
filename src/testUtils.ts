export type AssertTypeEqual<T, Expected> =
    T extends Expected
        ? (Expected extends T ? true : never)
        : never;