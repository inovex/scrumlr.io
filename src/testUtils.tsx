import {render, RenderOptions} from "@testing-library/react";
import {I18nextProvider} from "react-i18next";
import {FC, PropsWithChildren, ReactElement} from "react";
import {MemoryRouter} from "react-router";
import {Provider} from "react-redux";
import getTestStore from "utils/test/getTestStore";
import i18n from "./i18nTest";

export type AssertTypeEqual<T, Expected> = T extends Expected ? (Expected extends T ? true : never) : never;

const AllTheProvidersWithoutRouter: FC<PropsWithChildren> = ({children}) => (
  <I18nextProvider i18n={i18n}>
    <Provider store={getTestStore()}>{children}</Provider>
  </I18nextProvider>
);

const AllTheProviders: FC<PropsWithChildren> = ({children}) => (
  <MemoryRouter>
    <AllTheProvidersWithoutRouter>{children}</AllTheProvidersWithoutRouter>
  </MemoryRouter>
);

const customRender = (ui: ReactElement, options?: Omit<RenderOptions & {container: Element}, "wrapper">) => render(ui, {wrapper: AllTheProviders, ...options});
const customRenderWithoutRouter = (ui: ReactElement, options?: Omit<RenderOptions & {container: Element}, "wrapper">) =>
  render(ui, {wrapper: AllTheProvidersWithoutRouter, ...options});

export {customRender as render, customRenderWithoutRouter as renderWithoutRouter};
