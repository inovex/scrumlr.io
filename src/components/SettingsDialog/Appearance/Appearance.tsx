import classNames from "classnames";
import {useTranslation} from "react-i18next";
import {LanguageSettingsDropdown} from "../Components/LanguageSettingsDropdown";
import "../SettingsDialog.scss";
import "./Appearance.scss";
import {ThemeSettings} from "../Components/ThemeSettings";
import {NotificationSettings} from "../Components/NotificationSettings";
import {BoardReactionsSettings} from "../Components/BoardReactionsSettings";
import {SkinToneSelector} from "../Components/SkinToneSelector";

export const Appearance = () => {
  const {t} = useTranslation();

  return (
    <div className={classNames("settings-dialog__container", "accent-color__lean-lilac")}>
      <header className="settings-dialog__header">
        <h2 className="settings-dialog__header-text">{t("SettingsDialog.Appearance")}</h2>
      </header>
      <div className="appearance-container">
        <ThemeSettings />
        <NotificationSettings />
        <BoardReactionsSettings />
        <SkinToneSelector />
        <LanguageSettingsDropdown />
      </div>
    </div>
  );
};
