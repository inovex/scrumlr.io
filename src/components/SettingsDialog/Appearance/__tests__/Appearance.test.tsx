import "../matchMedia.mock";
import {Provider} from "react-redux";
import {render} from "testUtils";
import getTestStore from "utils/test/getTestStore";
import {Appearance} from "../Appearance";
import {getByLabelText} from "@testing-library/dom";

const createAppearance = () => (
  <Provider store={getTestStore()}>
    <Appearance />
  </Provider>
);

describe("Appearance", () => {
  test("should render all Settings correctly", () => {
    const {container} = render(createAppearance());
    expect(container.firstChild).toMatchSnapshot();
  });

  test("should have a notifications toggle", () => {
    const {container} = render(createAppearance());
    expect(getByLabelText(container, "Appearance.showHotkeyNotifications")).toBeInTheDocument();
  });
});
