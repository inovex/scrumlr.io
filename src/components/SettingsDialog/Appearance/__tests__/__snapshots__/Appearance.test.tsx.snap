// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appearance should render all Settings correctly 1`] = `
<div
  class="settings-dialog__container accent-color__lean-lilac"
>
  <header
    class="settings-dialog__header"
  >
    <h2
      class="settings-dialog__header-text"
    >
      SettingsDialog.Appearance
    </h2>
  </header>
  <div
    class="appearance-container"
  >
    <div
      class="appearance-settings__theme-container"
    >
      <span
        class="appearance-settings__theme-title"
      >
        Appearance.colorScheme
      </span>
      <form
        class="appearance-settings__theme-options"
      >
        <label
          class="appearence-settings__theme-option"
          for="auto"
          title="Appearance.SyncModeDescription"
        >
          <input
            checked=""
            id="auto"
            name="themeAuto"
            type="radio"
            value="auto"
          />
          <img
            alt="Appearance.colorScheme Auto"
            src="theme-preview-light.svg"
          />
          <img
            alt="Appearance.colorScheme Auto"
            src="theme-preview-dark.svg"
          />
          <p>
            <svg
              class="appearance-settings__theme-icon"
            >
              icon-settings.svg
            </svg>
            <span>
              Auto
            </span>
          </p>
        </label>
        <label
          class="appearence-settings__theme-option"
          for="light"
        >
          <input
            id="light"
            name="themeLight"
            type="radio"
            value="light"
          />
          <img
            alt="Appearance.colorScheme Appearance.colorSchemeLight"
            src="theme-preview-light.svg"
          />
          <p>
            <svg
              class="appearance-settings__theme-icon"
            >
              icon-lightmode.svg
            </svg>
            <span>
              Appearance.colorSchemeLight
            </span>
          </p>
        </label>
        <label
          class="appearence-settings__theme-option"
          for="dark"
        >
          <input
            id="dark"
            name="themeDark"
            type="radio"
            value="dark"
          />
          <img
            alt="Appearance.colorScheme Appearance.colorSchemeDark"
            src="theme-preview-dark.svg"
          />
          <p>
            <svg
              class="appearance-settings__theme-icon"
            >
              icon-darkmode.svg
            </svg>
            <span>
              Appearance.colorSchemeDark
            </span>
          </p>
        </label>
      </form>
    </div>
    <section>
      <button
        aria-checked="true"
        aria-label="Appearance.showHotkeyNotifications"
        class="settings-option-button"
        role="switch"
        type="button"
      >
        <span
          class="settings-option-button__label"
        >
          Appearance.showHotkeyNotifications
        </span>
        <div
          class="toggle toggle--active"
        />
      </button>
    </section>
    <section>
      <button
        aria-checked="true"
        aria-label="Appearance.showBoardReactions"
        class="settings-option-button"
        role="switch"
        type="button"
      >
        <span
          class="settings-option-button__label"
        >
          Appearance.showBoardReactions
        </span>
        <div
          class="toggle toggle--active"
        />
      </button>
    </section>
    <div
      class="settings-dropdown"
    >
      <button
        aria-controls="dropdown-list"
        aria-expanded="false"
        class="settings-dropdown__button"
        role="combobox"
      >
        <span>
          Appearance.EmojiSkinTone
        </span>
        <p
          class="settings-dropdown__item--current"
        >
          <span>
            👏
          </span>
          <svg
            class="settings-dropdown__item-icon settings-dropdown__item-icon--dropdown"
          >
            icon-arrow-next.svg
          </svg>
        </p>
      </button>
      <ul
        class="settings-dropdown__list"
        id="dropdown-list"
        role="listbox"
      >
        <li
          class="settings-dropdown__item"
        >
          <button
            class="settings-dropdown__button"
          >
            <span>
              👏
            </span>
          </button>
        </li>
        <li
          class="settings-dropdown__item"
        >
          <button
            class="settings-dropdown__button"
          >
            <span>
              👏🏻
            </span>
          </button>
        </li>
        <li
          class="settings-dropdown__item"
        >
          <button
            class="settings-dropdown__button"
          >
            <span>
              👏🏼
            </span>
          </button>
        </li>
        <li
          class="settings-dropdown__item"
        >
          <button
            class="settings-dropdown__button"
          >
            <span>
              👏🏽
            </span>
          </button>
        </li>
        <li
          class="settings-dropdown__item"
        >
          <button
            class="settings-dropdown__button"
          >
            <span>
              👏🏾
            </span>
          </button>
        </li>
        <li
          class="settings-dropdown__item"
        >
          <button
            class="settings-dropdown__button"
          >
            <span>
              👏🏿
            </span>
          </button>
        </li>
      </ul>
    </div>
    <div
      class="settings-dropdown"
    >
      <button
        aria-controls="dropdown-list"
        aria-expanded="false"
        class="settings-dropdown__button"
        role="combobox"
      >
        <span>
          Appearance.Language
        </span>
        <p
          class="settings-dropdown__item--current"
        >
          <svg
            class="settings-dropdown__item-icon"
          >
            US.svg
          </svg>
          <span>
            Language.english
          </span>
          <svg
            class="settings-dropdown__item-icon settings-dropdown__item-icon--dropdown"
          >
            icon-arrow-next.svg
          </svg>
        </p>
      </button>
      <ul
        class="settings-dropdown__list"
        id="dropdown-list"
        role="listbox"
      >
        <li
          class="settings-dropdown__item"
        >
          <button
            class="settings-dropdown__button"
          >
            <svg
              class="settings-dropdown__item-icon"
            >
              DE.svg
            </svg>
            <span>
              Language.german
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
